### ISSUE TITLE
Java14 Full Records Support Check Validation: UnusedImportsCheck

### ISSUE THREAD

**Issue by nmancus1** [checkstyle/checkstyle#8488]
Child of #8452 
Check documentation: https://checkstyle.sourceforge.io/config_imports.html#UnusedImports

From check documentation:
_Checks for unused import statements. Checkstyle uses a simple but very reliable algorithm to report on unused import statements. An import statement is considered unused if:_

- It is not referenced in the file. The algorithm does not support wild-card imports like import java.io.*;. Most IDE's provide very sophisticated checks for imports that handle wild-card imports.
- It is a duplicate of another import. This is when a class is imported more than once.
- The class imported is from the java.lang package. For example importing java.lang.String.
- The class imported is from the same package.
- Optionally: it is referenced in Javadoc comments. This check is on by default, but it is considered bad practice to introduce a compile time dependency for documentation purposes only. As an example, the import java.util.List would be considered referenced with the Javadoc comment {@link List}. The alternative to avoid introducing a compile time dependency would be to write the Javadoc comment as {@link java.util.List}.

```

➜  full-record-grammar /usr/lib/jvm/java-14-openjdk/bin/javac --enable-preview --source 14 TestClass.java
Note: TestClass.java uses preview language features.
Note: Recompile with -Xlint:preview for details.
➜  full-record-grammar cat config.xml    
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
      <module name="FinalParameters"/>
  </module>
</module>
➜  full-record-grammar cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
      <module name="UnusedImports"/>
  </module>
</module>
➜  full-record-grammar cat TestClass.java
import java.awt.Component; // ok
import java.util.List; // violation
import java.lang.*; // ok
import java.util.Iterator; // ok
import java.util.Enumeration; // violation
import java.util.Arrays; // ok
import javax.swing.JToolBar; // violation
import javax.swing.JToggleButton; // violation

import javax.swing.BorderFactory; // violation

class TestClass {
    public record MyRecord() {
        private static Object Component; // ok
        static Arrays arrays; // ok
        static Iterator<String> it; //ok
    }
}


➜  full-record-grammar java $RUN_LOCALE -jar ~/IdeaProjects/checkstyle/target/checkstyle-8.35-SNAPSHOT-all.jar -c config.xml TestClass.java
Starting audit...
[ERROR] /home/nick/Desktop/full-record-grammar/TestClass.java:2:8: Unused import - java.util.List. [UnusedImports]
[ERROR] /home/nick/Desktop/full-record-grammar/TestClass.java:5:8: Unused import - java.util.Enumeration. [UnusedImports]
[ERROR] /home/nick/Desktop/full-record-grammar/TestClass.java:7:8: Unused import - javax.swing.JToolBar. [UnusedImports]
[ERROR] /home/nick/Desktop/full-record-grammar/TestClass.java:8:8: Unused import - javax.swing.JToggleButton. [UnusedImports]
[ERROR] /home/nick/Desktop/full-record-grammar/TestClass.java:10:8: Unused import - javax.swing.BorderFactory. [UnusedImports]
Audit done.
Checkstyle ends with 5 errors.

```
This check works as intended, so I think we can close this issue.

**Comment by pbludov** [comment]
This check should be able to collect imports from the Javadoc of records and compact ctors:
```java
import java.util.Date;  // ok
import java.util.List;  // unexpected violation

/**
 * {@link Date}
 */
class TestClass {
}

/**
 * {@link List}
 */
record TestRecord() {
}
```
produces:
```
Starting audit...
[ERROR] /home/pbludov/src/tmp/TestClass.java:2:8: Unused import - java.util.List. [UnusedImports]
Audit done.
```

**Comment by esilkensen** [comment]
Fix is merged
